### PR TITLE
added_book-dto_and_mapping

### PR DESCRIPTION
**Pull request by SosunovychB**

### CODE REVIEW

File: src/main/java/book/store/service/BookService.java [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
+import book.store.dto.BookDto;
+import book.store.dto.CreateBookRequestDto;
 import java.util.List;
 
 public interface BookService {
-    Book save(Book book);
+    BookDto save(CreateBookRequestDto requestDto);
 
-    List<Book> findAll();
+    BookDto findById(Long id);
```
Comment (sokimaaa) [inline src/main/java/book/store/service/BookService.java:10]:
```suggestion
    Optional<BookDto> findById(Long id);
```

File: src/main/java/book/store/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -1,28 +1,41 @@
...
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id)
+        );
+        return bookMapper.toDto(book);
```
Comment (sokimaaa) [inline src/main/java/book/store/service/impl/BookServiceImpl.java]: Inline expression using Optional#map
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
                () -> new EntityNotFoundException("Can't find book with id: " + id)
        );
        return bookMapper.toDto(book);
```

File: src/main/java/book/store/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public Optional<BookDto> findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id)
+        );
+        return Optional.ofNullable(bookMapper.toDto(book));
```
Comment (sokimaaa) [inline src/main/java/book/store/service/impl/BookServiceImpl.java:33]: Inline expression using Optional#map
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
                () -> new EntityNotFoundException("Can't find book with id: " + id)
        );
        return Optional.ofNullable(bookMapper.toDto(book));
```

Comment (sokimaaa) [inline src/main/java/book/store/service/impl/BookServiceImpl.java]:
```suggestion
        return bookRepository.findById(id).map(...);
```

File: src/main/java/book/store/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id)
+        );
+    }
```
Comment (Rommelua) [inline src/main/java/book/store/controller/BookController.java]: This was not the correct comment, we should not return Optional from the Service layer.
image

Comment (SosunovychB) [reply src/main/java/book/store/controller/BookController.java]: Ok, I will return as it was before

**Pull request merged**